[2024-05-25T15:21:46.724+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-05-25T15:21:46.742+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: copy_single_csv_to_minio_dag.upload_single_file_to_minio scheduled__2022-01-01T00:03:00+00:00 [queued]>
[2024-05-25T15:21:46.750+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: copy_single_csv_to_minio_dag.upload_single_file_to_minio scheduled__2022-01-01T00:03:00+00:00 [queued]>
[2024-05-25T15:21:46.750+0000] {taskinstance.py:2306} INFO - Starting attempt 4 of 4
[2024-05-25T15:21:46.763+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): upload_single_file_to_minio> on 2022-01-01 00:03:00+00:00
[2024-05-25T15:21:46.767+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=298) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-05-25T15:21:46.769+0000] {standard_task_runner.py:63} INFO - Started process 299 to run task
[2024-05-25T15:21:46.769+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'copy_single_csv_to_minio_dag', 'upload_single_file_to_minio', 'scheduled__2022-01-01T00:03:00+00:00', '--job-id', '131', '--raw', '--subdir', 'DAGS_FOLDER/csv_copy_to_mino_dag.py', '--cfg-path', '/tmp/tmp823xwk9q']
[2024-05-25T15:21:46.770+0000] {standard_task_runner.py:91} INFO - Job 131: Subtask upload_single_file_to_minio
[2024-05-25T15:21:46.803+0000] {task_command.py:426} INFO - Running <TaskInstance: copy_single_csv_to_minio_dag.upload_single_file_to_minio scheduled__2022-01-01T00:03:00+00:00 [running]> on host 4c9a8a4e8975
[2024-05-25T15:21:46.858+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='copy_single_csv_to_minio_dag' AIRFLOW_CTX_TASK_ID='upload_single_file_to_minio' AIRFLOW_CTX_EXECUTION_DATE='2022-01-01T00:03:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-01-01T00:03:00+00:00'
[2024-05-25T15:21:46.859+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-05-25T15:21:46.966+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-05-25T15:21:46.967+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/csv_copy_to_mino_dag.py", line 37, in upload_single_file_to_minio
    client.upload_fileobj(data, bucket_name, dst)
  File "/home/airflow/.local/lib/python3.12/site-packages/boto3/s3/inject.py", line 642, in upload_fileobj
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/s3transfer/futures.py", line 103, in result
    return self._coordinator.result()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/s3transfer/futures.py", line 266, in result
    raise self._exception
  File "/home/airflow/.local/lib/python3.12/site-packages/s3transfer/tasks.py", line 139, in __call__
    return self._execute_main(kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/s3transfer/tasks.py", line 162, in _execute_main
    return_value = self._main(**kwargs)
                   ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/s3transfer/upload.py", line 764, in _main
    client.put_object(Bucket=bucket, Key=key, Body=body, **extra_args)
  File "/home/airflow/.local/lib/python3.12/site-packages/botocore/client.py", line 553, in _api_call
    return self._make_api_call(operation_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/botocore/client.py", line 1009, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidArgument) when calling the PutObject operation: S3 API Requests must be made to API port.
[2024-05-25T15:21:46.977+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=copy_single_csv_to_minio_dag, task_id=upload_single_file_to_minio, run_id=scheduled__2022-01-01T00:03:00+00:00, execution_date=20220101T000300, start_date=20240525T152146, end_date=20240525T152146
[2024-05-25T15:21:46.991+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 131 for task upload_single_file_to_minio (An error occurred (InvalidArgument) when calling the PutObject operation: S3 API Requests must be made to API port.; 299)
[2024-05-25T15:21:47.023+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-05-25T15:21:47.035+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-05-25T15:21:47.037+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
